### PR TITLE
feat(Accessibility): Associate error with field

### PR DESCRIPTION
**Pull request by thyhjwb6**
## Related issue
#990 

## Overview
When a form field is in an error state, the input needs aria-invalid and aria-describedby which is set to the ID of the div containing the error.

## Reason
Screen readers will not work when in an error state.

## Work carried out
- [x] Associate error with field
- [x] Refactor

### CODE REVIEW

**Review by m7kvqbe1 — APPROVED** [review]
LGTM

File: packages/react-component-library/src/enhancers/withFormik.tsx [hunk 1]
```diff
@@ -16,7 +17,7 @@ export const withFormik = (FormComponent: React.FC<any>) => ({
   className,
   ...props
 }: FormikProps) => {
-  const hasError = touched[field.name] && errors[field.name]
+  const hasError = !!(touched[field.name] && errors[field.name])
```
Comment (antictor) [inline packages/react-component-library/src/enhancers/withFormik.tsx]: Maybe not one for here but one comment I got from our designer was he didn't want to see errors until the first submit, but this is outside the scope of this

File: packages/react-component-library/src/enhancers/withFormik.tsx [hunk 2]
```diff
@@ -27,15 +28,25 @@ export const withFormik = (FormComponent: React.FC<any>) => ({
...
-        {...field}
-        {...props}
-        className={formComponentClassNames}
-      />
+      <FormComponent {...formComponentProps} />
       {hasError && (
-        <div className="rn-form__invalid-feedback" data-testid="error">
+        <div
+          className="rn-form__invalid-feedback"
+          data-testid="error"
```
Comment (antictor) [inline packages/react-component-library/src/enhancers/withFormik.tsx]: Can we include the field name in the test id? When I'm writing integration tests for a form I want to be able to test that an error for a particular field is raised and check it's contents.

File: packages/react-component-library/src/enhancers/withFormik.tsx [hunk 3]
```diff
@@ -27,15 +28,25 @@ export const withFormik = (FormComponent: React.FC<any>) => ({
     'is-invalid': hasError,
   })
 
+  const errorId = `${field.name}-error`
```
Comment (antictor) [inline packages/react-component-library/src/enhancers/withFormik.tsx]: Isn't there be a chance that two fields could have the same name?

File: packages/react-component-library/src/enhancers/withFormik.tsx (hunk 2 above)
Comment (thyhjwb6) [reply packages/react-component-library/src/enhancers/withFormik.tsx]: We are stripping out `data-testid` with babel. If this is Cypress could you do something like `<FormikField data-testid="id" />` and then `cy.get('[data-testid="id"]').siblings()`?

**Comment by m7kvqbe1** [comment]
re: Slack discussion

It's probably a good idea for us to spike the creation of some different forms to identify potential design concerns with form handling (across the library as a whole).

This looks good but we now have some nuance and complexity is growing. Consumption requires knowledge of both Formik and how we support its use. 

Dedicated documentation for 'forms' with working examples is needed.
